### ISSUE TITLE
C-c C-c does not behave as expected with emacs bindings

### ISSUE THREAD

**Issue by tobiasBora** [hlissner/doom-emacs#4258]
**What did you expect to happen?**

Usually, `C-c C-c` is a common shortcut in lots of modes (at least with emacs keybindings): in LaTeX it is used to compile, in Magit buffers it is used to validate the commit message... 

**What actually happened?**
However, in doom `C-c C-c`, seems to be bound to a prefix, and the commands that are available after looks pretty useless: `Esc` and `M-x`:
![image](https://user-images.githubusercontent.com/2164118/98922551-e04ff280-24d2-11eb-947f-f96776179f66.png)
which I could have use directly without typing `C-c C-c`. So I'm not sure why this could be useful, but in any case I'd much prefer to have `C-c C-c` bound to the old behaviour.

It is a bug due to the fact that most people here don't care about emacs keybindings and therefore don't need to use `C-c C-c`? If yes, would it be possible to move this prefix code during the configuration of evil?

Is there a quick method to have the "normal" behaviour?

**Steps to reproduce:**
1. Type `C-c C-c` in any buffer, for example in a LaTeX buffer, or you can also open magit, do a commit, and see that `C-c C-c` does not work (instead it is proposed to type `C-x #` but this shortcut is highly non natural to type, at least on a french keyboard)

**System information:**
<details><pre>
```
SYSTEM  type       gnu/linux
        config     x86_64-pc-linux-gnu
        shell      /run/current-system/sw/bin/bash
        uname      Linux 5.4.66 #1-NixOS SMP Thu Sep 17 11:47:56 UTC 2020 x86_64
        path       (/run/wrappers/bin ~/.nix-profile/bin /etc/profiles/per-user/$USER/bin /nix/var/nix/profiles/default/bin /run/current-system/sw/bin ~/.emacs.d/bin ~/.emacs.d/bin /nix/store/5qbf833ac8f0nisipgw6kp03n22a1rls-emacs-27.1/libexec/emacs/27.1/x86_64-pc-linux-gnu/)
EMACS   dir        ~/Documents/Informatique/Emacs_configurations/2020_doom/dot_emacs_dot_d/
        version    27.1
        build      nov. 12, 2020
        buildopts  --prefix=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-emacs-27.1 --disable-build-details --with-modules --with-x-toolkit=gtk3 --with-xft --with-cairo CFLAGS=-DMAC_OS_X_VERSION_MAX_ALLOWED=101200
        features   XPM JPEG TIFF GIF PNG RSVG CAIRO SOUND DBUS GSETTINGS GLIB NOTIFY INOTIFY LIBSELINUX GNUTLS LIBXML2 FREETYPE HARFBUZZ M17N_FLT LIBOTF ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 XDBE XIM MODULES THREADS LIBSYSTEMD JSON PDUMPER GMP
        traits     (server-running envvar-file)
DOOM    dir        ~/.doom.d/
        version    2.0.9
        build      grafted, HEAD -> develop, origin/develop, origin/HEAD 807f00a4 2020-11-02 23:56:43 -0500
        elc-files  0
        modules    (:completion company :ui doom doom-dashboard doom-quit (emoji +unicode) hl-todo minimap modeline ophints (popup +defaults) vc-gutter vi-tilde-fringe :editor file-templates fold snippets :emacs dired electric (undo +tree) vc :term vterm :checkers syntax (spell +flyspell) grammar :tools direnv (eval +overlay) lookup lsp magit :os tty :lang cc csharp emacs-lisp haskell (latex +latexmk +cdlatex +lsp +fold) lua markdown nix ocaml org rst scala sh :config (default +bindings))
        packages   ((debpaste) (terminal-here) (web-mode) (ido-completing-read+) (helm) (yasnippet-snippets))
        unpin      (n/a)
        elpa       (vterm yasnippet validate undo-tree avy auctex aggressive-indent)
```
</pre></details>

**Comment by hlissner** [comment]
![image](https://user-images.githubusercontent.com/510883/98985660-dc7f9880-24f1-11eb-9de8-790d86de5f7b.png)

I cannot reproduce this in vanilla Doom. `C-c C-c`, for instance, works as expected in magit, org-src blocks, and wgrep. As for latex:

![image](https://user-images.githubusercontent.com/510883/98985987-4e57e200-24f2-11eb-83b4-0984c0a0dcd1.png)

I suspect either:

1. An error occurred at startup; errors can short circuit Doom. Check out \*Messages\* (`C-h e` to open it) for errors or warnings post-startup.

2. Or you are binding your own `C-c` keybinds in your private config, causing conflicts. Please check your local config.

> It is a bug due to the fact that most people here don't care about emacs keybindings and therefore don't need to use C-c C-c?

When the evil module is disabled (as you have), our evil opinions are thrown out the window (by design).

**Comment by tobiasBora** [comment]
Thanks a lot for your answer. I investigated further, and I realized that the issue was completely on my side: I stupidly copy/pasted `smex`'s `use-package` code that was binding  `C-c C-c M-x` to the original `M-x` command... I'm so sorry to have lost your time :-\ Problem solved, and thanks a lot for your help!

> When the evil module is disabled (as you have), our evil opinions are thrown out the window (by design).

Great, good to know that there is a also support for normal emacs keybindings!
